### PR TITLE
Adding Mypy type hints

### PR DESCRIPTION
**Pull request by xydesa**

### CODE REVIEW

**Comment by xydesa** [comment]
@tdenewiler the mypy PR in statick (https://github.com/sscpac/statick/pull/202) removed support for pypy. Do you remember why? or how to test that support?

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -86,6 +86,10 @@ jobs:
           --exclude tests/.*/*valid_package/*
           src tests
 
+      - name: Mypy
+        run: |
+          mypy --ignore-missing-imports --allow-untyped-decorators --strict src/
```
Comment (tdenewiler) [inline .github/workflows/tests.yaml:91]:
```suggestion
          mypy --ignore-missing-imports --strict src/
```

The `--allow-untyped-decorators` is not needed here. It is used in Statick for the `deprecated` module.

File: README.md [hunk 2]
```diff
@@ -23,16 +24,20 @@ Custom exceptions can be applied the same way they are with
 
 The recommended method to install these Statick plugins is via pip:
 
-    pip install statick-md
+```shell
+pip install statick-md
```
Comment (tdenewiler) [inline README.md:28]:
```suggestion
python3 -m pip install statick-md
```

File: README.md [hunk 3]
```diff
@@ -50,7 +55,9 @@ In that case your directory structure will look like the following:
 
 To run with the default configuration for the statick-md tools use:
 
-    statick md-project/ statick-output/ --profile md-profile.yaml
+```shell
+statick md-project/ statick-output/ --profile md-profile.yaml
```
Comment (tdenewiler) [inline README.md:59]:
```suggestion
statick md-project/ --output-directory statick-output/ --profile md-profile.yaml
```

File: README.md [hunk 4]
```diff
@@ -68,7 +75,9 @@ This example will have custom exceptions in the md-project, such that the direct
 
 For this setup you will run the following:
 
-    statick md-project/ statick-output/ --user-paths md-project/statick-config/ --profile md-profile.yaml
+```shell
+statick md-project/ statick-output/ --user-paths md-project/statick-config/ --profile md-profile.yaml
```
Comment (tdenewiler) [inline README.md:79]:
```suggestion
statick md-project/ --output-directory statick-output/ --user-paths md-project/statick-config/ --profile md-profile.yaml
```

File: README.md [hunk 5]
```diff
@@ -87,7 +96,9 @@ The directory structure will look like:
 Using the example where we want to override the default exceptions with
 custom ones in the md-project, the command to run would be:
 
-    ./statick/statick md-project/ statick-output/ --user-paths statick-md/,md-project/statick-config/ --profile md-profile.yaml
+```shell
+./statick/statick md-project/ statick-output/ --user-paths statick-md/,md-project/statick-config/ --profile md-profile.yaml
```
Comment (tdenewiler) [inline README.md:100]:
```suggestion
./statick/statick md-project/ --output-directory statick-output/ --user-paths statick-md/,md-project/statick-config/ --profile md-profile.yaml
```

File: README.md [hunk 6]
```diff
@@ -99,10 +110,27 @@ future regressions) if you can add a small unit test which replicates the bug.
...
+
+Statick Markdown uses [mypy](http://mypy-lang.org/) to check that type hints are being followed properly.
+Type hints are described in [PEP 484](https://www.python.org/dev/peps/pep-0484/) and allow for static typing in Python.
+To determine if proper types are being used in Statick Markdown the following command will show any errors, and create several
+types of reports that can be viewed with a text editor or web browser.
+
+```shell
+python3 -m pip install mypy
+mkdir report
+mypy --ignore-missing-imports --allow-untyped-decorators --strict --html-report report/ --txt-report report src
```
Comment (tdenewiler) [inline README.md:123]:
```suggestion
mypy --ignore-missing-imports --strict --html-report report/ --txt-report report src
```

The `--allow-untyped-decorators` flag is not needed. It is used by Statick for the `deprecated` module.

File: README.md [hunk 7]
```diff
@@ -99,10 +110,27 @@ future regressions) if you can add a small unit test which replicates the bug.
...
+
 ### Formatting
 
 Statick code is formatted using [black](https://github.com/psf/black).
 To fix locally use
 
-    pip install black
-    black src tests
+```shell
+pip install black
```
Comment (tdenewiler) [inline README.md:134]:
```suggestion
python3 -m pip install black
```

**Review by tdenewiler — CHANGES_REQUESTED** [review]

**Review by tdenewiler — APPROVED** [review]

**Review by denewiler — APPROVED** [review]

**Pull request merged**